account_avatax_sale/models/sale_order.py

```diff
@@ -78,7 +78,10 @@ def _amount_all(self):
         """
         super()._amount_all()
         for order in self:
-            if order.tax_amount:
+            if not order.order_line:
+                order.onchange_reset_avatax_amount()
+            elif order.tax_amount:
+                order.avalara_compute_taxes()
```

Comment:
@metaminux This will trigger an API call on each computation.
You should not do this, it is against Avalara's policy for clients.
There is risk to get unexpected bills for the service or have requests rate limited and blocked.

Reply:
@dreispt Thank you for this review. Do you have any idea on how to fix this behaviour in another way so that payment amount is right when sending orders to customers ?
Besides, I can't find any documentation on rate limit for Avalara : https://developercommunity.avalara.com/s/question/0D55a00008028Z9CAI/avatax-api-rate-limit but I agree to do it another way if amounts are right at the end.

Reply:
@metaminux At That's a fair use case

Reply:
@metaminux For [certification requirements](https://developer.avalara.com/erp-integration-guide/sales-tax-badge/functional-review/): 

> The quote-to-cash process should specifically call out the times when the system is reaching out to AvaTax for a tax calculation and when the system is going to record a transaction to Avatax for further reporting and remittance.

 and 

> Show that a reasonable API request ratio is used during invoicing.
> Sales invoices or requests that use document type SalesInvoice should not call AvaTax multiple times for the same invoice.

---

You are describing a reasonable scenario.
Maybe we need some "Needs Tax Recomputation" dirty flag, and prevent "Send Mail" button or print "Sale Order" report on that case?